src/components/Contacts/ContactDetails/ContactDetailsTab/People/Items/EditContactOtherModal/EditContactOtherModal.tsx

```diff
@@ -0,0 +1,304 @@
+import {
+  ContactUpdateInput,
+  PreferredContactMethodEnum,
+} from '../../../../../../../../graphql/types.generated';
+import Modal from '../../../../../../common/Modal/Modal';
+import { ContactDetailsTabQuery } from '../../../ContactDetailsTab.generated';
+import { useApiConstants } from '../../../../../../Constants/UseApiConstants';
+import { useGetTimezones } from '../../../../../../../hooks/useGetTimezones';
+import { useUpdateContactOtherMutation } from './EditContactOther.generated';
+
+const ContactEditContainer = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  width: '100%',
+  flexDirection: 'column',
+  margin: theme.spacing(4, 0),
+}));
+
+const ContactInputWrapper = styled(Box)(({ theme }) => ({
+  position: 'relative',
+  padding: theme.spacing(0, 6),
+  margin: theme.spacing(2, 0),
+}));
+
+const LoadingIndicator = styled(CircularProgress)(({ theme }) => ({
+  margin: theme.spacing(0, 1, 0, 0),
+}));
+
+interface EditContactOtherModalProps {
+  contact: ContactDetailsTabQuery['contact'];
+  accountListId: string;
+  isOpen: boolean;
+  handleClose: () => void;
+}
+
+export const EditContactOtherModal: React.FC<EditContactOtherModalProps> = ({
+  accountListId,
+  contact,
+  isOpen,
+  handleClose,
+}): ReactElement<EditContactOtherModalProps> => {
+  const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const [
+    updateContactOther,
+    { loading: updating },
+  ] = useUpdateContactOtherMutation();
+
+  const constants = useApiConstants();
+  const languages = constants?.languages ?? [];
+  const timezones = useGetTimezones();
+
+  const contactOtherSchema: yup.SchemaOf<
+    Pick<
+      ContactUpdateInput,
+      | 'id'
+      | 'churchName'
+      | 'preferredContactMethod'
+      | 'locale'
+      | 'timezone'
+      | 'website'
+    >
+  > = yup.object({
+    id: yup.string().required(),
+    churchName: yup.string().nullable(),
+    preferredContactMethod: yup
+      .mixed<PreferredContactMethodEnum>()
+      .oneOf(Object.values(PreferredContactMethodEnum))
+      .nullable(),
+    locale: yup.string().nullable(),
+    timezone: yup.string().nullable(),
+    website: yup.string().nullable(),
+  });
+
+  const onSubmit = async (attributes: ContactUpdateInput) => {
+    try {
+      await updateContactOther({
+        variables: {
+          accountListId,
+          attributes: {
+            id: attributes.id,
+            churchName: attributes.churchName,
+            preferredContactMethod: attributes.preferredContactMethod,
+            locale: attributes.locale,
+            timezone: attributes.timezone,
+            website: attributes.website,
+          },
+        },
+      });
+      enqueueSnackbar(t('Contact updated successfully'), {
+        variant: 'success',
+      });
+    } catch (error) {
+      enqueueSnackbar(error.message, { variant: 'error' });
+      throw error;
+    }
+  };
+
+  const localizedContactMethod = (method: string) => {
```

Comment:
Is this function duplicate of https://github.com/CruGlobal/mpdx-react/pull/125/files#diff-7b58eb6ea9851a5d0a53bc6b3466a8c75bcfa397647971b2d5b312f195be87beR38?

Reply:
Yeah, I can probably leave it in ```ContactDetailsOther``` and then export it over to this component.

Reply:
Seems like you could still reuse most of it... `method ? localizedContactMethod(method) : 'N/A'`

Or call the version that returns `N/A` in both places knowing that branch will never get hit.